src/client/core/index.js

```diff
@@ -46,6 +47,7 @@ exports.promiseUtils           = promiseUtils;
 exports.textSelection          = textSelection;
 exports.waitFor                = waitFor;
 exports.delay                  = delay;
+exports.timeLimit              = timeLimit;
```

Comment:
I think `timeLimit` is not exactly what this function does. It's smth like `getTimeLimitedPromise`.